lib/taste_tester/commands.rb

```diff
@@ -210,5 +211,187 @@ def self.upload
+      end
+
+      changeset = BetweenMeals::Changeset.new(
+        logger,
+        repo,
+        start_ref,
+        nil, # compare with current working copy
+        {
+          :cookbook_dirs =>
+            TasteTester::Config.relative_cookbook_dirs,
+          :role_dir =>
+            TasteTester::Config.relative_role_dir,
+        },
+        @track_symlinks,
+      )
+
+      return changeset
+    end
+
+    def self._find_impact(changeset)
+      if TasteTester::Config.relative_cookbook_dirs.length > 1
+        logger.error('Knife deps does not support multiple cookbook paths.')
+        logger.error('Flatten the repository or use the custom_impact hook.')
+        exit(1)
+      end
+
+      cookbooks = Set.new(changeset.cookbooks)
+      roles = Set.new(changeset.roles)
+
+      if cookbooks.empty? && roles.empty?
+        logger.error('No cookbooks or roles have been modified.')
+        exit(1)
+      end
+
+      unless cookbooks.empty?
+        logger.warn('Modified Cookbooks:')
+        cookbooks.each { |cb| logger.warn("\t#{cb}") }
+      end
+      unless roles.empty?
+        logger.warn('Modified Roles:')
+        roles.each { |r| logger.warn("\t#{r}") }
+      end
+
+      return _find_impact_roles(cookbooks, roles)
+    end
+
+    def self._find_impact_roles(cookbooks, roles)
+      # Now that we have a list of changed cookbooks and roles, we can check
+      # which roles contain modified dependencies. For simplicity, we will
+      # check using brute force by iterating through each role and comparing
+      # its dependencies with the set of modified files. If a match exists,
+      # the role was impacted, otherwise we can move on.
+      # This may change in the future if it is a perfomance bottleneck.
+
+      role_dir = TasteTester::Config.role_dir
+      chef_dir = File.join(TasteTester::Config.repo,
+                           TasteTester::Config.base_dir)
+
+      # CLI options for knife call
+      config = "--config #{TasteTester::Config.knife_config}"
+      chef_path = "--chef-repo-path #{chef_dir}"
+      options = '--tree --recurse'
+
+      # shell out to knife once and parse the resulting file for dependents
+      # if knife did not exit with 0, print whatever it returned and exit
+      logger.warn('Finding dependencies (this may take a minute or two)...')
+      knife = Mixlib::ShellOut.new(
+        "knife deps /#{role_dir}/*.rb #{options} #{config} #{chef_path}",
+      )
+      knife.run_command
+      if knife.error?
+        knife.invalid!('knife deps failed, please check output below')
+      end
+
+      # create a hash between roles and their dependencies
+      logger.warn('Processing dependencies...')
+      deps_tree = _build_dependency_tree(knife.stdout)
+
+      # search the hash for roles containing any modified files
+      logger.warn('Searching for impacted roles...')
+
+      impact_roles = []
+      roles.each { |r| impact_roles |= r.name }
```

Comment:
did you mean `impact_roles = roles.map { |r| r.name }` ?

Reply:
I still want `impact_roles` to be a set. Is there a better way than `impact_roles = Set.new(roles.map { |r| r.name })`?